spec/features/saml/ial1_sso_spec.rb

```diff
@@ -70,22 +70,34 @@
       )
     end
 
-    it 'after session timeout, signing in takes user back to SP' do
+    it 'after session timeout, signing in takes user back to SP', :js, allow_browser_log: true do
+      allow(IdentityConfig.store).to receive(:session_check_delay).and_return(0)
+
       user = create(:user, :fully_registered)
       request_url = saml_authn_request_url
 
       visit request_url
       sp_request_id = ServiceProviderRequestProxy.last.uuid
+      fill_in_credentials_and_submit(user.email, user.password)
+
+      Warden.on_next_request do |proxy|
+        proxy.env['devise.skip_trackable'] = true
+        session = proxy.env['rack.session']
+        session['warden.user.user.session']['last_request_at'] = User.timeout_in.ago.to_i - 1
```

Comment:
Chatting a bit with @mitchellhenke on Slack, I suspect that it's not a problem in production only because the `last_request_at` is being extended by calls to `/active`. That seems very unreliable, and is improved in #7966 since (a) the “active” poll request [doesn’t extend the timeout](https://github.com/18F/identity-idp/blob/298ce7b5a52e1b1ad866cdf0ef07547d171b47d6/app/controllers/api/internal/sessions_controller.rb#L7) and (b) we don’t rely on the session at the time of timeout to maintain the `request_id`, instead [maintaining it by assigning the URL at page load](https://github.com/18F/identity-idp/pull/7966/files#diff-d9f0af6637696cfed376657cae1d3583fbf160df9e285a23d62dbdff00f31604).